tide_core.drush.inc

```diff
@@ -0,0 +1,188 @@
+<?php
+
+/**
+ * @file
+ * Drush file to define custom drush command redirect-import.
+ */
+
+use Drupal\redirect\Entity\Redirect;
+
+/**
+ * Implements hook_drush_command().
+ */
+function tide_core_drush_command() {
+  $items = [];
+
+  $items['redirect-import'] = [
+    'description' => "Imports a file and adds redirects to the redirect table",
+    'arguments' => [
+      'filepath' => 'Path to CSV (comma-separated), JSON or PHP file containing old and new URLs for redirects',
+    ],
+  ];
+
+  return $items;
+
+};
+
+/**
+ * Get the node id of the page to redirect to.
+ */
+function get_node_id($to) {
+  if ((substr($to, 0, 4) == 'http')) {
+    return $to;
+  }
+  $to = (substr($to, 0, 1) == '/') ? $to : '/' . $to;
+  $path = \Drupal::service('path.alias_manager')->getPathByAlias($to);
+  if (preg_match('/node\/(\d+)/', $path, $matches)) {
+    return 'node/' . $matches[1];
+  }
+  return $to;
+}
+
+/**
+ * Uses redirect entity to save entry to DB.
+ */
+function add_redirects($from, $to) {
+  if ($from != '' && $to != '') {
+    $redirectrepo = redirect_repository();
+    if ($redirectrepo->findBySourcePath($from)) {
+      return;
+    }
+    $vicgov = $to == 'https://vic.gov.au' || $to == 'http://vic.gov.au' || $to == 'vic.gov.au';
+    $wwwvicgov = $to == 'https://www.vic.gov.au' || $to == 'http://www.vic.gov.au' || $to == 'www.vic.gov.au';
+    if ($vicgov || $wwwvicgov) {
+      $to = 'site-4/';
```

Comment:
So `$to` is, I guess, a path prefix for certain URLs? Maybe that should just be a lookup in an array of redirect exceptions rather than the conditional statements above?

Reply:
Ah no, looking at this, it is rewriting the whole `$to` value :+1: